packages/testcontainers/src/container-runtime/clients/container/container-client.ts

```diff
@@ -22,7 +22,7 @@ export interface ContainerClient {
   stop(container: Container, opts?: { timeout: number }): Promise<void>;
   attach(container: Container): Promise<Readable>;
   logs(container: Container, opts?: ContainerLogsOptions): Promise<Readable>;
-  exec(container: Container, command: string[], opts?: { log: boolean }): Promise<ExecResult>;
+  exec(container: Container, command: string[], opts?: Partial<ExecOptions>): Promise<ExecResult>;
```

Comment:
Asking without being a Typescript person: would this be a risk for the existing users, as a breaking change? If they were already passing an opts object including the log boolean, what will happen to them at compile and run time? Because we are introducing here a new type for the ExecOptions (which makes sense to me in any case). Do you think this can cause a BC?

Reply:
Note that I added two different `ExecOptions` types in this PR. 

One, in the public API that does not include the `log` parameter, and another that supports the `log` option in the container client interface (this case above 👆🏼).

So, TL;DR, these changes are backwards compatible 🙂

Reply:
I agree that  is following the same pattern that we have to other kind of types like `ExecResult`.  So in the internal API we keep log prop without breaking risk at we had without this PR.

I agree to add test and, at least to me, is a LGTM.